### PR TITLE
Disposals bin nanoUI

### PR DESCRIPTION
**Pull request by PJB3005**

### CODE REVIEW

File: code/modules/recycling/disposal.dm [hunk 1]
```diff
@@ -23,434 +23,422 @@
...
-
-			air_contents = new/datum/gas_mixture()
-			//gas.volume = 1.05 * CELLSTANDARD
-		update()
+// create a new disposal
+// find the attached trunk (if present) and init gas resvr.
+/obj/machinery/disposal/New()
+	..()
+	spawn(5)
+		trunk = locate() in loc
```
Comment (d3athrow) [inline code/modules/recycling/disposal.dm]: funny story, for(var/thing in list) is cheaper than locate.

File: code/modules/recycling/disposal.dm [hunk 2]
```diff
@@ -23,434 +23,422 @@
...
+			GM.attack_log += "<span class='warning'> [user]([user.ckey]) has attempted to put [GM]([GM.ckey]) in disposals.</span>"
+			for (var/mob/V in viewers(usr))
+				V.show_message("[usr] starts putting [GM.name] into the disposal.", 3)
+			if(do_after(usr, 20))
+				if (GM.client)
+					GM.client.perspective = EYE_PERSPECTIVE
+					GM.client.eye = src
+				GM.loc = src
+				for (var/mob/C in viewers(src))
+					C.show_message("<span class='warning'>[GM.name] has been placed in the [src] by [user].</span>", 3)
```
Comment (d3athrow) [inline code/modules/recycling/disposal.dm]: Should be changed to visible message

Comment (ComicIronic) [inline code/modules/recycling/disposal.dm]: As should the case above this one.

File: code/modules/recycling/disposal.dm (hunk 1 above)
Comment (PJB3005) [inline code/modules/recycling/disposal.dm]: I didn't even code this, I only made it absolute but fine.

File: code/modules/recycling/disposal.dm (hunk 2 above)
Comment (PJB3005) [inline code/modules/recycling/disposal.dm]: Again I didn't code this, but okay.

**Comment by d3athrow** [comment]
Should be good to merge when the freeze is over. Though I do feel bad for making more things use nano when theres a handful of people who just cannot use nano for whatever reason. Some weird fucking settings they've got somehow.

**Pull request merged**